src/backend/joanie/core/models/courses.py

```diff
@@ -212,6 +217,43 @@ def get_abilities(self, user):
             "manage_accesses": is_owner_or_admin,
         }
 
+    def signature_backend_references_to_sign(
+        self, contracts_ids: list[str] | None = None
+    ):
+        """
+        Return the list of references that should be signed by the organization.
+        """
+        contracts_ids_filter = Q()
+        if contracts_ids:
+            contracts_ids_filter = Q(id__in=contracts_ids)
+        contracts_to_sign = list(
+            Contract.objects.filter(
+                contracts_ids_filter,
+                signature_backend_reference__isnull=False,
+                submitted_for_signature_on__isnull=False,
+                order__organization=self,
```

Comment:
What do you think to ensure also that the student has signed ? 

```suggestion
                signature_backend_reference__isnull=False,
                submitted_for_signature_on__isnull=False,
                student_signed_on__isnull=False,
                order__organization=self,
```